test/runtests.jl

```diff
@@ -66,36 +66,35 @@ Generate vector spaces for the layer-internal indices of a MERA of `meratype`, t
 """
 function random_internalspaces(extspaces, meratype)
     width = (meratype == TernaryMERA ? 3 :
-             meratype in (BinaryMERA, ModifiedBinaryMERA) ? 2 : nothing)
-    intspaces = tuple((begin
-                           # The fuse just removes nested product state structure.
-                           Vext = fuse(extspaces[i])
-                           Vnext = i == length(extspaces) ? Vext : extspaces[i+1]
-                           sects = Dict()
-                           for s in sectors(Vext)
-                               d = dim(Vext, s)
-                               rand(Bool) && d > 1 && (d = d - 1)
-                               sects[s] = d
-                           end
-                           Vint = spacetype(Vext)(sects...)
-                           # Check that we didn't accidentally make the isometric part
-                           # non-isometric. If we did, just make Vint = Vext.
-                           Vint_fusion = fuse(reduce(⊗, repeat([Vint], width)))
-                           infinum(Vint_fusion, Vnext) != Vnext && (Vint = Vext)
-                           Vint
-                       end
-                       for i in 1:length(extspaces))...)
+    meratype in (BinaryMERA, ModifiedBinaryMERA) ? 2 : nothing)
+    intspaces = ntuple(length(extspaces)) do i
+        # The fuse just removes nested product state structure.
+        Vext = fuse(extspaces[i])
+        Vnext = i == length(extspaces) ? Vext : extspaces[i+1]
+        sects = Dict()
+        for s in sectors(Vext)
+            d = dim(Vext, s)
+            rand(Bool) && d > 1 && (d = d - 1)
+            sects[s] = d
+        end
+        Vint = spacetype(Vext)(sects...)
+        # Check that we didn't accidentally make the isometric part
+        # non-isometric. If we did, just make Vint = Vext.
+        Vint_fusion = fuse(fill(Vint, width)...)
+        infimum(Vint_fusion, Vnext) != Vnext && (Vint = Vext)
+        Vint
+    end
     return intspaces
 end
 
 """
 Test type stability, and type stability only, of various methods.
 """
-function test_type_stability(meratype, spacetype)
+function test_type_stability(::Type{meratype}, ::Type{S}) where {meratype, S}
```

Comment:
We could also replace all `meratype`s with just `M`. I can do this later too.

Reply:
As you wish; I used `S` because `spacetype` conflicted with TensorKit's `spacetype` method, while testing things interactively in the REPL